reframe/utility/sanity.py

```diff
@@ -610,11 +616,26 @@ def extractall(patt, filename, tag=0, conv=None, encoding='utf-8'):
         Group ``0`` refers always to the whole match.
         Since the file is processed line by line, this means that group ``0``
         returns the whole line that was matched.
-    :arg conv: A callable that takes a single argument and returns a new value.
-        If provided, it will be used to convert the extracted values before
-        returning them.
-    :returns: A list of the extracted values from the matched regex.
+    :arg conv: A callable or iterable of callables taking a single argument
+        and returning a new value.
+        If provided, and is not an iterable it will be used to convert
+        the extracted values for all the capturing groups of ``tag``
+        returning the converted values.
+        If an iterable of callables is provided, each one will be used to
+        convert the corresponding extracted capturing group of `tag`.
```

Comment:
```suggestion
        Otherwise each conversion function will be used to
        convert the value extracted from the corresponding capturing group in `tag`.
```